README.md

```diff
@@ -1,95 +1,269 @@
+- [Using Jigawatt](#using-jigawatt)
+  - [Basic Usage](#basic-usage-example)
+  - [Middleware Structure](#middleware-structure)
+    - [awesomize](#awesomize)
+    - [io](#io)
+    - [transform](#transform)
+    - [Summary](#summary)
+- [Contribute](#contribute)
+- [License](#license)
+
 ## Installation
-`npm install jigawatt`
+`npm install jigawatt --save`
+
+## Using Jigawatt
+
+First, require the module.
 
-## Middleware Structure
+`const JW = require('jigawatt')`
 
-### Properties
-Jigawatt Middleware must consist of at least one of three properties:
-- `awesomize`: `(Validator -> AwesomizeSpec) -> Request -> Object a`
-  - validates/sanitizes the request to initialize the `data` object
-  - _for more info on `awesomize`, see https://www.npmjs.com/package/awesomize_
+For this walkthrough, we'll also be using Ramda.
 
-- `io`: `Request, Data -> Object a`
-  - returns an object that is merged into `req.data`
+`const R = require('ramda')`
 
-- `transform`: `Request, Data -> Object a`
-  - returns an object to become the new `req.data` value
+### Basic Usage Example
+
+For this example, we have a poll and poll responses stored in a database. Consider the following endpoint:
 
-### Example Middleware
 ```
-const _         = require('ramda')
-const Bluebird  = require('bluebird')
+app.get('/best-city-results/:id', (req, res) => {
+  const data       = { pollId : req.params.id }
+      , formatPoll = JW(pollDetails, data)
+
+  formatPoll(data, { json : (data) => data })
+    .then((details) => // Do something with result
+    .catch((err)    => // Handle the error
+})
 
-const Order = require('../domain/order.js')
+...
+```
 
-const getById = {
+Jigawatt middleware functions, such as `pollDetails` in this case, are created to sanitize, validate, and/or normalize the incoming data (`awesomize`), then use that awesomized data to query the database (`io`), and lastly, format the results to our liking and return the output (`transform`).
+
+```
+const pollDetails = {
   awesomize: (v) => ({
-    order_id: {
-      read: _.path(['params', 'orderId'])
-    , validation: [ v.required ]
+    pollId : {
+      sanitize : R.toLower
+    , validate : isCorrectLength
     }
   })
 
-  io: (req, data) => {
-    return Bluebird.props({
-      order : Order.getById(data.order_id)
-    })
-  }
-};
-
-const uniteDetails = {
-  transform: (req, data) => {
-    return {
-      order     : data.order
-    , customer  : data.customer
-    , product   : data.product
-    , shipping  : data.shipping  
-    }
-  }
+, io: (req, data) => ({
+    poll      : db.Poll.findOne({ _id: data.pollId })
+  , responses : db.Vote.find({ poll_id: data.pollId })
+  })
+
+, transform: (req, data) => ({
+    poll    : data.poll.title
+  , results : tallyVotes(
+                getCity(data.poll)
+              , getAnswer(data.responses)
+              )
+  })
 }
+```
+
+Our Jigawatt middleware returns a thenable, which can be used to handle the resulting data as desired:
```

Comment:
Just a plain object, to get the thenable you need to do JW.promisify... I think.